app/forms/tasks/base_form.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Tasks
+  class BaseForm
+    include ActiveModel::Model
+    include Rails.application.routes.url_helpers
+
+    attr_reader :task, :case_record, :enforcement
+    delegate :parent, to: :task
+
+    def initialize(task)
+      @task = task
+      @case_record = @task.case_record
+      @enforcement = @case_record.caseable
```

Comment:
Safe for the time being but can't hurt to do the same check here that we do in the controller:

```suggestion
      @enforcement = @case_record.caseable if case_record.enforcement?
```

Reply:
Feels like this should be a `before_action` and not littered through the code.

Reply:
I mean a ` before_action` in the tasks controller as that's where it's initially loaded.